src/highlight.js

```diff
@@ -702,15 +703,19 @@ const HLJS = function(hljs) {
 
     if (shouldNotHighlight(language)) return;
 
-    // support for v10 API
     fire("before:highlightElement",
       { el: element, language: language });
 
+    if (!options.ignoreUnescapedHTML && element.innerHTML !== element.textContent) {
```

Comment:
```html
<div id="foo">
  I'm just a &lt;code&gt;sample&lt;/code&gt; block.
</div>

<div id="innerHTML"></div>
<div id="textContent"></div>
<div id="result"></div>

<script>
document.getElementById("innerHTML").innerText = document.getElementById("foo").innerHTML;
document.getElementById("textContent").innerText = document.getElementById("foo").textContent;
document.getElementById("result").innerText = document.getElementById("foo").innerHTML == document.getElementById("foo").textContent;
</script>
```

https://jsfiddle.net/uj5qzn2e/

Am I doing something wrong? Or does `innerHTML` need to be have HTML entities transformed before they can be compared with `textContent`

Reply:
Oh, my test cases locally didn't include escaped HTML... that certainly breaks the comparison approach.  I think instead perhaps we just need to check node.children...